autoextract_poet/providers.py

```diff
@@ -0,0 +1,49 @@
+from typing import Type
+
+from scrapy.http import Request
+from scrapy.statscollectors import StatsCollector
+from scrapy_poet.page_input_providers import (
+    PageObjectInputProvider,
+    provides,
+)
+
+from autoextract_poet.inputs import (
+    AutoExtractProductResponseData,
+    AutoExtractProductListResponseData,
+)
+from autoextract_poet.utils import request
+
+
+class AutoExtractProvider(PageObjectInputProvider):
+
+    page_type: str
+    page_input_class: Type
+
+    def __init__(self, request: Request, stats: StatsCollector):
+        self.request = request
+        self.stats = stats
+
+    async def __call__(self):
+        try:
+            # FIXME: how to configure if you want full HTML or not?
+            data = await request(self.request.url, self.page_type)
```

Comment:
Since we're counting stats for success/error, maybe count a global one for requests?
This way, we can cross check the numbers, success + errors = global

Reply:
Done.